src/main/java/no/digipost/signature/client/security/KeyStoreConfig.java

```diff
@@ -83,15 +88,47 @@ public PrivateKey getPrivateKey() {
+     * certificate using {{@link #fromOrganizationCertificate(InputStream, String)}}
+     */
+    @Deprecated
+    public static KeyStoreConfig fromKeyStore(final InputStream javaKeyStore, final String alias, final String keyStorePassword, final String privatekeyPassword) {
+        return fromJavaKeyStore(javaKeyStore, alias, keyStorePassword, privatekeyPassword);
+    }
+
+    /**
+     * Create a {@link KeyStoreConfig} from a Java Key Store containing an Organization Certificate (Virksomhetssertifikat).
+     *
+     * @param javaKeyStore       A stream of the certificate in JCEKS format.
+     * @param alias              The alias of the organization certificate in the key store.
+     * @param keyStorePassword   The password for the key store itself.
+     * @param privatekeyPassword The password for the private key of the organization certificate within the key store.
+     * @return The config, containing the certificate, the private key and the certificate chain.
+     */
+    public static KeyStoreConfig fromJavaKeyStore(final InputStream javaKeyStore, final String alias, final String keyStorePassword, final String privatekeyPassword) {
+        KeyStore ks = KeyStoreType.JCEKS.loadKeyStore(javaKeyStore, keyStorePassword);
+        return new KeyStoreConfig(ks, alias, keyStorePassword, privatekeyPassword);
+
+    }
+
+    /**
+     * Create a {@link KeyStoreConfig} from an Organization Certificate (Virksomhetssertifikat).
+     *
+     * @param organizationCertificateStream A stream of the certificate in PKCS12 format. The file should have .p12-file ending.
+     * @param privatekeyPassword            The password for the private key of the organization certificate.
+     * @return The config, containing the certificate, the private key and the certificate chain.
+     */
+    public static KeyStoreConfig fromOrganizationCertificate(final InputStream organizationCertificateStream, final String privatekeyPassword) {
+        KeyStore ks = KeyStoreType.PKCS12.loadKeyStore(organizationCertificateStream, privatekeyPassword);
         try {
-            KeyStore ks = KeyStore.getInstance("JCEKS");
-            ks.load(keyStore, keyStorePassword.toCharArray());
-            return new KeyStoreConfig(ks, alias, keyStorePassword, privatekeyPassword);
-        } catch (FileNotFoundException e) {
-            throw new KeyException("Failed to initialize key store. Are you sure the file exists?", e);
-        } catch (KeyStoreException | NoSuchAlgorithmException | IOException | java.security.cert.CertificateException e) {
+            Enumeration aliases = ks.aliases();
+            String keyName = (String) aliases.nextElement();
+            return new KeyStoreConfig(ks, keyName, privatekeyPassword, privatekeyPassword);
+        } catch (KeyStoreException e) {
             throw new KeyException("Failed to initialize key store", e);
+        } catch (NoSuchElementException e) {
```

Comment:
Kan man her kanskje fange _både_ KeyStoreException og NoSuchElementException? Begge deler skjer jo i forbindelse med at man forsøker å hente ut nøkkel-aliaser. Foreslår å rett og slett flytte KeyStoreException sammen med NoSuchElementException her, og bare bruke den exception'en som kastes under her. `new KeyException("Failed to initialize key store", e)` virker uansett litt utdatert, da KeyStore'en _er_ initialisert.